frontend/ui-pablo/app/components/Organisms/AllBondTable.tsx

```diff
@@ -76,55 +72,65 @@ export const AllBondTable: React.FC = () => {
           </TableRow>
         </TableHead>
         <TableBody>
-          {pools.map((pool, index) => (
+          {bondOffers.slice(0, count).map((bond, index) => (
             <TableRow
               key={index}
-              onClick={handleBondClick}
+              onClick={() => handleBondClick(bond.offerId.toString())}
               sx={{ cursor: "pointer" }}
             >
               <TableCell align="left">
-                {pool.token2 ? (
+                {bond.principalAsset && (bond.principalAsset as any).baseAsset && (bond.principalAsset as any).quoteAsset ? (
                   <PairAsset
                     assets={[
-                      { icon: pool.token1.icon, label: pool.token1.symbol },
-                      { icon: pool.token2.icon, label: pool.token2.symbol },
+                      {
+                        icon: (bond.principalAsset as any).baseAsset.icon,
```

Comment:
@easteregg yeah however, it does have a type defined if you look into useBondOffer hook. i need to use any because vs typescript lint confuses the types